contracts/crypto-verify/src/msg.rs

```diff
@@ -0,0 +1,38 @@
+#![allow(clippy::field_reassign_with_default)] // see https://github.com/CosmWasm/cosmwasm/issues/685
+
+use cosmwasm_std::{Binary, Deps, StdResult};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct InitMsg {}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum HandleMsg {
+    VerifySignature {
+        /// Sha-256 hash of the message to verify (32 bytes).
+        message_hash: Binary,
```

Comment:
Here we could take the unhashed `message` and do the hashing in the contract. This better demos the end use case where hash functions end up as dependencies of the contract.